crates/rpc-client/src/pathfinder/proofs.rs

```diff
@@ -106,6 +106,9 @@ impl PathfinderClassProof {
     ///
     /// However, the proof should always start with the root node, which means all we have
     /// to do is hash the first node in the proof to get the same thing.
+    ///
+    /// NOTE: the v0.8 RPC spec does NOT require the proof to be in order, in which case it is
+    ///       much trickier to guess what the root node is.
```

Comment:
To elaborate here:

The proof is (1) not required to be in order by the spec and (2) is not in order in my experimentation in `v0.8`. Nor should it be since it consolidates proofs.

A workaround would be to reconstruct a tree view of the proof, in which case it should be clear what the root node is, but this is definitely not ideal.

The `v0.8` RPC proof endpoint does provide commitments, but there are caveats.